multichain-api-gateway/src/lib.rs

```diff
@@ -0,0 +1,201 @@
+use std::{net::TcpListener, str};
+
+use actix_web::{dev::Server, web, App, HttpRequest, HttpResponse, HttpServer};
+use futures::{stream, StreamExt};
+use reqwest::Client;
+use serde_json::Value;
+use url::Url;
+
+use crate::config::{BlockScoutSettings, Instance, Settings};
+
+pub mod config;
+
+fn build_urls(path: &str, query: &str, settings: &BlockScoutSettings) -> Vec<(Instance, Url)> {
+    settings
+        .instances
+        .iter()
+        .map(|Instance(net, subnet)| {
+            let mut url = settings.base_url.clone();
+            url.set_path(&format!("{}/{}{}", net, subnet, path));
+            url.set_query(Some(query));
+            (Instance(net.clone(), subnet.clone()), url)
+        })
+        .collect::<Vec<_>>()
+}
```

Comment:
I suggest to create urls (without query part) only once on service initialization. You can do that by creating a new structure (e.g. `Gateway`, but I'm not sure about naming) with the following layout: 
```
struct Gateway {
    apis: Vec<Url>,
    concurrent_requests: usize,
}
```
and which would be initialized from `BlockScoutSettings` via implementing `TryFrom` trait. You may initialize that structure in the `run`, so that url parsing would happen only once and shutdown the server if any variant could not result in valid url